pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
We definitely do not make API calls in validation - we would do so in the
strategy at worst.

On Mon, May 23, 2016 at 2:41 PM, Eric Chiang <notifications@github.com>
wrote:

> In pkg/registry/clusterrolebinding/policybased/storage.go
> <https://github.com/kubernetes/kubernetes/pull/25634#discussion_r64263097>
> :
>
> > @@ -0,0 +1,59 @@
> > +/*
>
> @erictune <https://github.com/erictune> after more thinking, I think the
> inherent promise that role bindings don't apply at the cluster level gets
> us most of this behavior we want.
>
> The mutability of roles, means we have to address @olegshaldybin
> <https://github.com/olegshaldybin>'s comments. And in this case, I'm
> having trouble figuring out how to centralize all of this logic since
> there's no good example of existing storage validation requiring API access.
>
> Making these checks an admission controller means we have to deal with
> @liggitt <https://github.com/liggitt>'s comments of unintentional
> escalation, but it also gets us the localhost without escalation checks we
> talked about earlier.
>
> Wondering if there's some way we can deal with this after we merge the API
> group and authorizer.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25634/files/82fa2c3b354a84fa2f9be6d9420701df6b36fe2b..17bffb41bd5a5687d7bda2d30f26159071e51cab#r64263097>
>